api/metadata/v1/metadata_sdk.go

```diff
@@ -0,0 +1,108 @@
+package metadata
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+var (
+	metadataURL = "http://169.254.42.42"
+)
+
+// API metadata API
+type API struct {
+}
+
+// NewAPI returns a API object from a Scaleway client.
+func NewAPI() *API {
+	return &API{}
+}
+
+// Metadata returns the metadata avialable from the server
+func (*API) Metadata() (*Metadata, error) {
+	resp, err := http.Get(metadataURL + "/conf?format=json")
+	if err != nil {
+		return nil, err
+	}
+
+	metadata := &Metadata{}
+	err = json.NewDecoder(resp.Body).Decode(metadata)
+	if err != nil {
+		return nil, err
```

Comment:
should return a [SdkError](https://github.com/scaleway/scaleway-sdk-go/blob/master/scw/errors.go#L8)

Reply:
fixed